mama/c_cpp/src/c/dqstrategyplugin.c

```diff
@@ -0,0 +1,927 @@
+/*
+ * dqstrategyplugin.c
+ *
+ *  Created on: 25 Oct 2017
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "mama/mama.h"
+#include "mama/status.h"
+#include "mama/types.h"
+#include "dqstrategyplugin.h"
+#include "mama/msgstatus.h"
+#include "mama/reservedfields.h"
+#include "mama/timer.h"
+#include "subscriptionimpl.h"
+#include "imagerequest.h"
+#include "dqstrategy.h"
+#include "transportimpl.h"
+#include "bridge.h"
+#include "msgutils.h"
+#include "mama/statfields.h"
+#include "mama/statscollector.h"
+
+#include "msgimpl.h"
+#include "refreshtransport.h"
+/**
+ * Implementations should subclass this method and call queueLimitExceeded if
+ * a message queue overflows.
+ */
+
+#define strategyImpl ((dqStrategyImpl*)(strategy))
+#define DQSTRATEGY_PLUGIN_NAME "dqstrategy"
+
+static int handleEvent (mamaPluginInfo pluginInfo, mamaTransport transport);
```

Comment:
This is a minor niggle (and I know I wrote this one) but could you change this to

static int
handleEvent() ...ect

Just to match the style of the rest of the file